### PR TITLE
Example K8s-grafana deployment

### PR DESCRIPTION
**Pull request by mariomac**
To be merged after https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/32

Simple deployment to demonstrate the instrumentation without requiring any extra launcher.

### CODE REVIEW

File: deployments/grafana-demo/02-auto-instrument.yml [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+      port: 4318
+      protocol: TCP
+      targetPort: http-traces
+    - name: grpc
+      port: 4317
+      protocol: TCP
+      targetPort: grpc-traces
+---
+apiVersion: apps/v1
+kind: DaemonSet
```
Comment (edeNFed) [inline deployments/grafana-demo/02-auto-instrument.yml:132]: Why not deploy it as a sidecar container in the goblog deployment? 
Deploying it as a DaemonSet would only instrument a single process even if you deploy multiple instances of the goblog application.

File: deployments/grafana-demo/02-auto-instrument.yml [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+      }
+    }
+      
+    otelcol.exporter.otlp "tempo" {
+      // Send traces to a locally running Tempo without TLS enabled.
+      client {
+        endpoint = env("TEMPO_ENDPOINT")
+        auth     = otelcol.auth.basic.creds.handler
+      }
+    }
```
Comment (dineshg13) [inline deployments/grafana-demo/02-auto-instrument.yml:40]: Can we use local graphana or prometheus ? In opentelemetry-demo, we use same https://github.com/open-telemetry/opentelemetry-demo/blob/5f7517aad8e2f1365c7d4b173194bfdba2da4489/docker-compose.yml#L624

**Review by dineshg13 — CHANGES_REQUESTED** [review]
Can we rewrite example to use deployed version of graphana or prometheus rather than remote one

File: deployments/grafana-demo/00-demo-app.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    matchLabels:
+      app: goblog
+  template:
+    metadata:
+      labels:
+        app: goblog
+    spec:
+      containers:
+        - name: goblog
+          image: mariomac/goblog:dev
```
Comment (dineshg13) [inline deployments/grafana-demo/00-demo-app.yml:21]: How is this image build ?

**Comment by mariomac** [comment]
Closing this PR as, thinking it twice, it has too many issues to be part of the standard documentation/examples. (e.g. the demo application is just a personal container)

**Pull request closed without merging**